management/roles.py

```diff
@@ -93,22 +94,68 @@ class Aura_Teller(Assassin):
+                        mail = Mailbox().respond("{} - <@{}> appears to you as a **Threat**!".format(player.emoji,player.id),me.channel)
+                        mail.respond("This means they're part of the wolf pack!",me.channel)
+                        return mail.log("The **Aura Teller** <@{}> has inspected <@{}>, a **{}**, and discovered them as a threat!".format(me.id,player.id,player.role.name))
+                    Mailbox().respond("{} - <@{}> does **not** appear to you as a threat.".format(player.emoji,player.id),me.channel)
+                    return mail.log("The **Aura Teller** <@{0}> has inspected <@{1}>. As <@{1}> is a **{2}**, they did not appear as a threat to them.".format(me.id,player.id,player.role.name))
+            return Mailbox().respond("Sorry, who could tell that I couldn't find your target?",me.channel)
+        return Mailbox().respond("Sorry, buddy. You can't use your powers right now.",me.channel)
 
 # ===============================================
 class Baker(Innocent):
     
     def __init__(self):
         self.name = "Baker"
-        self.killers = ["Innocent", "Assassin", "Barber", "Cult Leader", "Executioner", "Huntress", "Witch", "Werewolf", "Lone Wolf", "Devil", "Wager", "Horseman", "Pyromancer"]
 
 # ===============================================
 class Butcher(Innocent):
     
     def __init__(self):
         self.name = "Butcher"
-        self.killers = ["Innocent", "Assassin", "Barber", "Cult Leader", "Executioner", "Huntress", "Witch", "Werewolf", "Lone Wolf", "Devil", "Wager", "Horseman", "Pyromancer"]
+
+# ===============================================
+class Barber(Innocent):
+
+    def __init__(self):
+        self.name == "Barber"
+    
+    def night(self,me):
+        me.uses = 0
+        me.votes = 1
+        me.threatened = 0
+    
+    def day(self,me):
+        me.uses = 1
+        me.fakerole = self.name
+        me.bitten = False
+    
+    def power(self,me,victim):
+        if me.undead == True:
+            return Mailbox().respond("You're undead! You can't use your powers!",me.channel)
+        if me.uses > 0:
+            if me.id == victim.id:
+                return Mailbox().respond("Whether suicide is an option or not is for you to find out. However, it is not in this game. Try again, please.",me.channel)
+            for player in Game_Control().participants:
+                if player.id == victim.id and player.role.name not in ["Spectator", "Dead"]:
+                    mail = Mailbox().log("The **Barber** <@{}> has chosen to assassinate".format(me.id))
+                    if player.souls > 0:
+                        mail.log_add(" <@{}>, a soulless **{}** who happened to have a soul protecting them.".format(player.id,player.role.name))
+                        mail.respond("Though you thought you had cut up your target nicely enough, it seems they have somehow survived! Too bad...",me.channel)
+                        return mail
```

Comment:
What about runner? The thing? Or one of the other roles who can survive killing?

Reply:
Those do not survive a barber's attack. As of now, the only way to survive a barber's  attack is by having received a soul from the Devil.